File: Examples/ObjcDemo/RixSDKPodDemo/RixEngineObjcDemo/UI/Max/MaxBannerVC.m
#import "MaxBannerVC.h"
#import <AppLovinSDK/AppLovinSDK.h>
#import "AdConfig.h"

@interface MaxBannerVC () <MAAdViewAdDelegate>

@property (nonatomic, strong) UILabel *label;
@property (nonatomic, assign) BOOL isLoading;
@property (nonatomic, strong) MAAdView *bannerView;
@property (nonatomic, copy) NSString *TAG;

@end

@implementation MaxBannerVC

- (void)viewDidLoad {
    [super viewDidLoad];
    self.TAG = @"Max-banner:";
    self.view.backgroundColor = [UIColor whiteColor];
    self.navigationItem.title = NSLocalizedString(@"max_banner", @"");

    UIButton *bnLoad = [self createButtonWithTitle:NSLocalizedString(@"load_ad", @"") action:@selector(buttonLoad)];
    [self.view addSubview:bnLoad];

    self.label = [self createLabel];
    [self.view addSubview:self.label];

    self.bannerView = [[MAAdView alloc] initWithAdUnitIdentifier:[AdConfig Max_Banner_Ad_Id]];
    self.bannerView.translatesAutoresizingMaskIntoConstraints = NO;
    [self.view addSubview:self.bannerView];

    CGFloat height = [MAAdFormat banner].adaptiveSize.height;

    [NSLayoutConstraint activateConstraints:@[
        [bnLoad.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [bnLoad.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [bnLoad.heightAnchor constraintEqualToConstant:50],
        [bnLoad.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor],

        [self.label.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [self.label.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [self.label.heightAnchor constraintEqualToConstant:50],
        [self.label.topAnchor constraintEqualToAnchor:bnLoad.bottomAnchor constant:20],

        [self.bannerView.topAnchor constraintEqualToAnchor:self.label.bottomAnchor constant:20],
        [self.bannerView.leadingAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.leadingAnchor],
        [self.bannerView.trailingAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.trailingAnchor],
        [self.bannerView.heightAnchor constraintEqualToConstant:height],
    ]];
}

- (void)buttonLoad {
    NSLog(@"load ad");
    if (self.isLoading) {
        return;
    }
    [self updateUI:YES message:NSLocalizedString(@"loading", @"")];
    [self loadAd];
}

- (void)loadAd {
    self.bannerView.delegate = self;
    [self.bannerView setExtraParameterForKey:@"adaptive_banner" value:@"true"];
    [self.bannerView setLocalExtraParameterForKey:@"adaptive_banner_width" value:@400];
    [self.bannerView loadAd];
}

- (void)updateUI:(BOOL)loading message:(NSString *)msg {
    self.isLoading = loading;
    self.label.text = msg;
}

#pragma mark - MAAdViewAdDelegate

- (void)didExpandAd:(MAAd *)ad {
    NSLog(@"%@ didExpand", self.TAG);
}

- (void)didCollapseAd:(MAAd *)ad {
    NSLog(@"%@ didCollapse", self.TAG);
}

- (void)didLoadAd:(MAAd *)ad {
    NSLog(@"%@ didLoad", self.TAG);
    [self updateUI:NO message:NSLocalizedString(@"load_success", @"")];
}

- (void)didFailToLoadAdForAdUnitIdentifier:(NSString *)adUnitIdentifier withError:(MAError *)error {
    NSString *msg = [NSString stringWithFormat:@"%ld: %@", (long)error.code, error.description];
    NSLog(@"%@ didFailToLoadAd: %@", self.TAG, msg);
    [self updateUI:NO message:[NSString stringWithFormat:NSLocalizedString(@"load_failed", @""), msg]];
}

- (void)didDisplayAd:(MAAd *)ad {
    NSLog(@"%@ didDisplay", self.TAG);
}

- (void)didHideAd:(MAAd *)ad {
    NSLog(@"%@ didHide", self.TAG);
}

- (void)didClickAd:(MAAd *)ad {
    NSLog(@"%@ didClick", self.TAG);
}

- (void)didFailToDisplayAd:(MAAd *)ad withError:(MAError *)error {
    NSLog(@"%@ didFail: %ld %@", self.TAG, (long)error.code, error.description);
}

@end

File: Examples/SwiftDemo/RixSDKPodDemo/AlxAdsDemo/UI/Max/MaxBannerVC.swift
//
//  MaxBannerVC.swift
//  AdsDemo
//
//  Created by liu weile on 2023/6/2.
//

import UIKit
import AppLovinSDK

class MaxBannerVC: BaseUIViewController {
    
    private let TAG = "Max-banner:"
    
    private var label:UILabel!

    private var isLoading:Bool=false

    private var bannerView:MAAdView!

    override func viewDidLoad() {
        super.viewDidLoad()

        view.backgroundColor = .white
        navigationItem.title=NSLocalizedString("max_banner", comment: "")

        let bnLoad=createButton(title: NSLocalizedString("load_ad", comment: ""),  action: #selector(buttonLoad))
        view.addSubview(bnLoad)
        
        label=createLabel()
        view.addSubview(label)

        bannerView=MAAdView(adUnitIdentifier: AdConfig.Max_Banner_Ad_Id)
        bannerView.translatesAutoresizingMaskIntoConstraints=false
        view.addSubview(bannerView)
        let height:CGFloat=MAAdFormat.banner.adaptiveSize.height
          

        NSLayoutConstraint.activate([
            bnLoad.leftAnchor.constraint(equalTo: view.leftAnchor),
            bnLoad.rightAnchor.constraint(equalTo: view.rightAnchor),
            bnLoad.widthAnchor.constraint(equalTo: view.widthAnchor),
            bnLoad.heightAnchor.constraint(equalToConstant: 50),
            bnLoad.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),


            label.leftAnchor.constraint(equalTo: view.leftAnchor),
            label.rightAnchor.constraint(equalTo: view.rightAnchor),
            label.widthAnchor.constraint(equalTo: view.widthAnchor),
            label.heightAnchor.constraint(equalToConstant: 50),
            label.topAnchor.constraint(equalTo: bnLoad.bottomAnchor, constant: 20),

//            bannerView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
            bannerView.topAnchor.constraint(equalTo: label.bottomAnchor, constant: 20),
            bannerView.widthAnchor.constraint(equalTo: view.safeAreaLayoutGuide.widthAnchor),
            bannerView.heightAnchor.constraint(equalToConstant: height),

        ])
    }
    
    
    @objc func buttonLoad(){
        print("load ad")
        if isLoading {
            return
        }
        updateUI(true, NSLocalizedString("loading", comment: ""))

        loadAd()
    }
    
    private func loadAd(){
        bannerView.delegate=self
        
        bannerView.setExtraParameterForKey("adaptive_banner", value: "true")
        bannerView.setLocalExtraParameterForKey("adaptive_banner_width", value: 400)
//        bannerView.getAdFormat().getAdaptiveSize(400).getHeight() // Set your ad height to this value
        
        bannerView.loadAd()
    }
    
    func updateUI(_ loading:Bool,_ msg:String){
        self.isLoading=loading
        self.label.text=msg
    }

   

}

extension MaxBannerVC:MAAdViewAdDelegate{
    
    func didExpand(_ ad: MAAd) {
        print("\(TAG) didExpand")
    }
    
    func didCollapse(_ ad: MAAd) {
        print("\(TAG) didCollapse")
    }
    
    func didLoad(_ ad: MAAd) {
        print("\(TAG) didLoad")
        updateUI(false, NSLocalizedString("load_success", comment: ""))
    }
    
    func didFailToLoadAd(forAdUnitIdentifier adUnitIdentifier: String, withError error: MAError) {
        let msg="\(error.code): \(error.description)"
        print("\(TAG) didFailToLoadAd: \(msg)")
        updateUI(false, String(format: NSLocalizedString("load_failed", comment: ""), msg))
    }
    
    func didDisplay(_ ad: MAAd) {
        print("\(TAG) didDisplay")
    }
    
    func didHide(_ ad: MAAd) {
        print("\(TAG) didHide")
    }
    
    func didClick(_ ad: MAAd) {
        print("\(TAG) didClick")
    }
    
    func didFail(toDisplay ad: MAAd, withError error: MAError) {
        print("\(TAG) didFail: \(error.code) \(error.description)")
    }
    
    
}
